scripts/localization.swift

```diff
@@ -0,0 +1,349 @@
+        guard let tokenRegex = twnTokenRegex else {
+            return ""
+        }
+        return self.replacingMatches(fromRegex: tokenRegex, withFormat: "%%%@$@")
+    }
+    
+    var twnNativeLocalization: String {
+        guard let tokenRegex = iOSTokenRegex else {
+            return ""
+        }
+        return self.replacingMatches(fromRegex: tokenRegex, withFormat: "$%@")
+    }
+}
+
+func writeStrings(fromDictionary dictionary: [String: String], toFile: String, escaped: Bool) throws {
+    var output = ""
+    let sortedDictionary = dictionary.sorted(by: { (kv1, kv2) -> Bool in
+        return kv1.key < kv2.key
+    })
+    for (key, value) in sortedDictionary {
+        output.append("\"\(key)\" = \"\(escaped ? value.escapedString : value)\";\n")
+    }
+    
+    try output.write(toFile: toFile, atomically: true, encoding: .utf8)
+}
+
+
+func exportLocalizationsFromSourceCode() {
+	let basePath = "Wikipedia/iOS Native Localizations/Base.lproj/Localizable.strings"
+	let qqqPath = "Wikipedia/Localizations/qqq.lproj/Localizable.strings"
+	let enPath = "Wikipedia/Localizations/en.lproj/Localizable.strings"
+	guard let baseDictionary = NSDictionary(contentsOfFile: basePath) else {
+	       print("ABORTING")
+	       abort()
+	}
+
+	var qqq = [String: String]()
+	var en = [String: String]() 
+
+	do {
+	   let commentSet = CharacterSet(charactersIn: "/* ")
+	   let quoteSet = CharacterSet(charactersIn: "\"")
+	   let string = try String(contentsOfFile: basePath)
+	   let lines = string.components(separatedBy: .newlines)
+	   var currentComment: String?
+	   var currentKey: String?
+	   var commentsByKey = [String: String]()
+	   for line in lines {
+	       let cleanedLine = line.trimmingCharacters(in: .whitespaces)
+	       if cleanedLine.hasPrefix("/*") {
+	           currentComment = cleanedLine.trimmingCharacters(in: commentSet)
+	           currentKey = nil
+	       } else if currentComment != nil {
+	           let quotesRemoved = cleanedLine.trimmingCharacters(in: quoteSet)
+
+	           if let range = quotesRemoved.range(of: "\" = \"") {
+	               currentKey = quotesRemoved.substring(to: range.lowerBound)
+	           }
+	       }
+	       if let key = currentKey, let comment =  currentComment {
+	           commentsByKey[key] = comment
+	       }
+	   }
+
+	   for (key, comment) in commentsByKey {
+	       qqq[key] = comment.twnNativeLocalization
+	   }
+	   try writeStrings(fromDictionary: qqq, toFile: qqqPath, escaped: false)
+
+	   for (key, value) in baseDictionary {
+	       guard let value = value as? String, let key = key as? String  else {
+	           continue
+	       }
+	       en[key] = value.twnNativeLocalization
+	   }
+	   try writeStrings(fromDictionary: en, toFile: enPath, escaped: true)
+	} catch let error {
+	   print("error: \(error)")
+	}
+}
+
+
+func importLocalizationsFromTWN() {
+	let basePath = "Wikipedia/iOS Native Localizations/Base.lproj/Localizable.strings"
```

Comment:
Here is where localizations are imported from twn - scan each language, convert tokens from $1, $2, $3, etc to %1$@, %2$@, %3$@, etc and generate plural dictionaries from `{{PLURAL:`